library/Faktory/Ent/Batch/Status.hs

```diff
@@ -33,8 +34,14 @@ data BatchStatus = BatchStatus
 newtype ReadCustomBatchId = ReadCustomBatchId
   { _bid :: BatchId
   }
-  deriving stock Generic
-  deriving anyclass FromJSON
+  deriving stock (Show,Eq,Generic)
+
+instance FromJSON ReadCustomBatchId where
+  -- Faktory seems to use the key '_bid' when enqueuing callback jobs and 'bid' for normal jobs...
+  parseJSON v = withParser "_bid" v <|> withParser "bid" v
+   where
+    withParser s =
+      withObject "ReadCustomBatchId" $ \o -> ReadCustomBatchId <$> o .: s
```

Comment:
Just mentioning because this has bitten me before: using `(<|>)` on `Parser` means the final failure is what will be reported when both sides fail. So in this case, it'd be like "Cannot find `bid`". That could confuse someone who is expecting `_bid` to be looked for, but this seems super minor to me so it's probably fine.

If I wanted to avoid it, I might write it with `(<|>)` on `Maybe` and a custom `fail`.

```hs
withObject "ReadCustomBatchId" $ \o -> do
  mBid <- liftA2 (<|>) <$> o .: "_bid" <*> o .: "bid"
  maybe (fail "_bid nor bid found...") pure mBid
```

Reply:
👍 you're right that this could be a confusing error, but we end up `hush`ing it in `jobBatchId` anway.

Is your formulation (minus the nice error message) the same as: `withObject "ReadCustomBatchId" $ \o -> o .: "_bid" <|> o .: "bid"`?  I know that formulation "consumes" the object on the first `.:` and  then always fails the second `.:`

Reply:
Also, aeson has a function to attach a custom message. So you could take either of your form and just put your own message on it -- as another option for avoiding the confusion.

Reply:
I'm not sure why, but when testing having both parsers underneath a single `withObject` I was getting `"parsing ReadCustomBatchId failed, expected Object, but encountered String"` when the first `.:` failed and the second `.:` was tried.

Reply:
Hmm, I'd have to poke at a real example. `withObject ".." $ \o -> o .: "this" <|> o .: "that"` should work fine -- assuming the precedence is as I expect.